src/Dialogs/Weather/RASPDialog.cpp

```
@@ -24,13 +45,170 @@
+    ContourSpacing(contour_density, height_scale);
+  renderer.GenerateImage(false, height_scale,
+                         0, 0, Angle::Zero(), contour_spacing);
+
+  if (use_alpha) {
+    // Draw checkerboard background for alpha styles
+    constexpr unsigned CHECK_SQUARES_Y = 7;
+    const unsigned check_squares_x = CHECK_SQUARES_Y * width / bar_height;
+    const unsigned check_size_x = width / check_squares_x;
+    const unsigned check_size_y = bar_height / CHECK_SQUARES_Y;
+    const Color light_color(230, 230, 230);
+    const Color dark_color(26, 26, 26);
```

Cppcheck (2.21.0)

[error] Dereference of an invalid iterator

(derefInvalidIterator)